features/ui/src/main/java/org/opennms/alec/rest/ALECRestImpl.java

```diff
@@ -1,26 +1,171 @@
+    private final BundleContext bundleContext;
+    private final ObjectMapper objectMapper;
+
+    public ALECRestImpl(DataStore dataStore, BundleContext bundleContext) {
+        this.dataStore = dataStore;
+        this.bundleContext = bundleContext;
+        objectMapper = new ObjectMapper();
+    }
 
     @Override
     public Response ping() {
-        // How do I identify the user?
+        return Response.ok("pong!!").build();
+    }
 
+    @Override
+    public Response getConfigurations() {
+        LOG.debug("Get Configurations");
+        ConfigurationImpl.Builder configuration = ConfigurationImpl.newBuilder();
+        KeyEnum.stream().forEach(keyEnum -> {
+            String value = (String) dataStore.get(keyEnum.toString(), ALEC_CONFIG).orElse("");
+            switch (keyEnum) {
+                case ENGINE:
+                    try {
+                        configuration.engineParameter(objectMapper.readValue(value, EngineParameter.class));
+                    } catch (JsonProcessingException e) {
+                        configuration.engineParameter(null);
+                    }
+                    break;
+                case AGREEMENT:
+                default:
+                    configuration.keyValue(KeyValueImpl.newBuilder().keyEnum(keyEnum).value(value).build());
+            }
+        });
+        return Response.ok().entity(configuration.build()).build();
+    }
 
+    @Override
+    public Response getConfiguration(KeyEnum keyEnum) {
+        LOG.debug("Get Configuration {}", keyEnum);
+        String value = (String) dataStore.get(keyEnum.toString(), ALEC_CONFIG).orElse("");
+        switch (keyEnum) {
+            case ENGINE:
+                try {
+                    return Response.ok().entity(objectMapper.readValue(value, EngineParameter.class)).build();
+                } catch (JsonProcessingException e) {
+                    return somethingWentWrong(e);
+                }
+            case AGREEMENT:
+            default:
+                return Response.ok().entity(value).build();
+        }
+    }
 
-        return Response.ok("pong!!").build();
+    @Override
+    public Response setConfiguration(KeyValue keyValue) {
+        LOG.debug("\n=============================================\n" +
+                "Set Configuration: {}\n" +
+                "=============================================", keyValue.toString());
+        return Response.ok().entity(dataStore.put(keyValue.getKeyEnum().toString(), keyValue.getValue(), ALEC_CONFIG)).build();
     }
 
     @Override
-    public Response handleWebhook(String body) {
-        LOG.debug("Got payload: {}", body);
-        return Response.ok().build();
+    public Response setEngineConfiguration(EngineParameter engineParameter) {
+        try {
+            //Retrieve Driver, only one driver is registered
+            ServiceReference<?>[] engineRegistryRefs = bundleContext.getAllServiceReferences(EngineRegistry.class.getCanonicalName(), null);
```

Comment:
If there is only one implementation fo EngineRegistry ( at least that's what I see currently).  Can you make it directly available with service reference ?
` <service ref="driver" interface="org.opennms.alec.engine.api.EngineRegistry"/>`